src/main/java/org/prebid/server/floors/BasicPriceFloorAdjuster.java

```diff
@@ -34,17 +44,69 @@ private static ExtRequestBidadjustmentfactors extractBidadjustmentfactors(BidReq
         return ObjectUtil.getIfNotNull(extPrebid, ExtRequestPrebid::getBidadjustmentfactors);
     }
 
+    private static ImpMediaType chooseMediaType(Imp imp, ExtRequestBidadjustmentfactors extBidadjustmentfactors, String bidder) {
+        if (MapUtils.isEmpty(extBidadjustmentfactors.getMediatypes())) {
+            return null;
+        }
+
+        final List<ImpMediaType> availableMediatypesForImp = retrieveImpMediatypes(imp);
+        if (CollectionUtils.isEmpty(availableMediatypesForImp)) {
+            return null;
+        }
+
+        final TreeMap<BigDecimal, ImpMediaType> factorToMediatype = new TreeMap<>();
+
+        for (ImpMediaType mediaType : availableMediatypesForImp) {
+            final Map<String, BigDecimal> mediaTypeValues = extBidadjustmentfactors.getMediatypes().get(mediaType);
+            if (MapUtils.isEmpty(mediaTypeValues)) {
+                continue;
+            }
+
+            final BigDecimal factor = mediaTypeValues.get(bidder);
+            if (factor == null) {
+                continue;
+            }
+            factorToMediatype.put(factor.setScale(4, RoundingMode.UNNECESSARY), mediaType);
+        }
+
+        return MapUtils.isEmpty(factorToMediatype) ? null : factorToMediatype.firstEntry().getValue();
+    }
+
+    private static List<ImpMediaType> retrieveImpMediatypes(Imp imp) {
+        final List<ImpMediaType> availableMediatypes = new ArrayList<>();
+
+        if (imp.getAudio() != null) {
```

Comment:
Minor. Just for consistency with project, place it in order (banner, video, native, audio)